src/swapCost/index.ts

```diff
@@ -55,12 +51,9 @@ export class SwapCostCalculator {
                 tokenAddress
             );
 
-            const tokenDecimals = await this.getTokenDecimals(tokenAddress);
-
             // Coingecko returns price of token in terms of ETH
             // We want the price of 1 ETH in terms of the token base units
-            const ethPerTokenWei = scale(bnum(ethPerToken), 18 - tokenDecimals);
-            const ethPriceInToken = BONE.div(ethPerTokenWei).dp(0);
+            const ethPriceInToken = bnum(1).div(bnum(ethPerToken));
```

Comment:
Can we revert these changes and perform necessary scaling into human values before passing the resulting value into `optimizeSwapAmounts`? With the work in #148, we'd ideally be able to eventually remove usage of `bignumber.js` entirely so I'd like to confine it to particular areas of the codebase if possible.

Reply:
Aren't we already using the same bignumber.js functions in the previous version, i.e. bnum and div?
Scaling to human values before `optimizeSwapAmounts` means we need to get the token decimals from somewhere which isn't readily available as things currently stand. Plus the return amount isn't used anywhere else as far as I know so wouldn't it be clearer to return the value in the format we need it?

Reply:
Fair point about the existing implementation using bnum (this is no longer the case in #148 however), what I mean is that I'd like to avoid using human readable values wherever possible as this we currently need to do a ton of scaling (with associated computational + mental load). 

My point is eventually `optimizeSwapAmounts` can use EVM values so I think it would be best if we had the scaling that we need to do now to happen near there so we can remove it once `optimizeSwapAmounts` no longer needs to take human values.